storage/wdpost_run.go

```diff
@@ -750,33 +753,18 @@ func (s *WindowPoStScheduler) submitPost(ctx context.Context, proof *miner.Submi
 	return sm, nil
 }
 
-func (s *WindowPoStScheduler) setSender(ctx context.Context, msg *types.Message, spec *api.MessageSendSpec) {
+func (s *WindowPoStScheduler) setSender(ctx context.Context, msg *types.Message, spec *api.MessageSendSpec) error {
 	mi, err := s.api.StateMinerInfo(ctx, s.actor, types.EmptyTSK)
 	if err != nil {
-		log.Errorw("error getting miner info", "error", err)
-
-		// better than just failing
-		msg.From = s.worker
-		return
-	}
-	worker, err := s.api.StateAccountKey(ctx, mi.Worker, types.EmptyTSK)
-	if err != nil {
-		log.Errorw("error getting account key", "error", err)
-
-		msg.From = s.worker
-		return
-	}
-	// worker should keep sync with state on chain, if not, will met error when estimate msg gas.
-	if s.worker != worker {
-		s.worker = worker
-		msg.From = s.worker
+		return xerrors.Errorf("error getting miner info: %w", err)
 	}
```

Comment:
My original patch didn't bother caching (see above comment) but this was added in by a subsequent commit on master.  I can add this back if desired, but it seems unnecessary and storing the key seems error prone.